pkg/orchestration/wiring/ec2compute/common/common.go

```diff
@@ -59,19 +66,54 @@ type StateComputeSpec struct {
 	RenameEnvVar map[string]string `json:"rename,omitempty"`
 }
 
-func generateSecretEnvVarsResource(stateResource *orch_v1.StateResource, computeSpec *StateComputeSpec, dependencyReferences []smith_v1.Reference) (smith_v1.Resource, string, error) {
+func generateSecretResource(compute voyager.ResourceName, envVars map[string]string, dependencyReferences []smith_v1.Reference) (smith_v1.Resource, error) {
+	objectName := wiringutil.MetaNameWithPostfix(compute, secretPluginNamePostfix)
+
+	// if this is for pods then we just have data
+	envVarJSONString, err := json.Marshal(map[string]map[string]string{
+		inputParameterEnvVarName: envVars,
+	})
+	if err != nil {
+		return smith_v1.Resource{}, errors.WithStack(err)
+	}
+
+	secretPluginSpec, err := runtime.DefaultUnstructuredConverter.ToUnstructured(&secretplugin.Spec{
+		Data: map[string][]byte{
+			envVarOutputSecretKey: envVarJSONString,
+		},
+	})
+	if err != nil {
+		return smith_v1.Resource{}, errors.WithStack(err)
+	}
+
+	instanceResource := smith_v1.Resource{
+		Name:       smith_v1.ResourceName(objectName),
+		References: dependencyReferences,
+		Spec: smith_v1.ResourceSpec{
+			Plugin: &smith_v1.PluginSpec{
+				Name:       secretplugin.PluginName,
+				ObjectName: objectName,
+				Spec:       secretPluginSpec,
+			},
+		},
+	}
+
+	return instanceResource, nil
+}
+
+func generateSecretEnvVarsResource(compute voyager.ResourceName, renameEnvVar map[string]string, dependencyReferences []smith_v1.Reference) (smith_v1.Resource, error) {
 	// We use objectName for both the smith resource name and the kubernetes metadata name,
 	// since there's only one of these per state resource (no possibility of clash).
-	objectName := fmt.Sprintf(secretEnvVarFormatString, stateResource.Name)
+	objectName := wiringutil.MetaNameWithPostfix(compute, secretEnvVarNamePostfix)
```

Comment:
This is old-ish code, need to use both helper functions, not just this one. And it would be better to inline the calls to be more explicit about where this thing is used. i.e. the name is not supposed to be re-used most of the time.

Reply:
And the comment can go too.

Reply:
I'm deleting this in my next PR so I'm just going to leave it